Makefile.core.mk

```diff
@@ -264,10 +256,23 @@ gen-code: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and
 
 .PHONY: gen-charts
 gen-charts: ## Pull charts from maistra/istio repository
-	hack/download-charts.sh v${MINOR_VERSION} https://github.com/${ISTIO_REPOSITORY} ${ISTIO_COMMIT_30}
+	@yq eval 'to_entries | .[] | "./hack/download-charts.sh " + .key + " " + .value.repo + " " + .value.commit + " " + ((.value.charts // []) | join(" "))' < versions.yaml | sh
+	@hack/copy-crds.sh "resources/$$(yq eval 'keys | .[0]' versions.yaml)/charts"
+
+.PHONY: update-version-list
```

Comment:
Generally speaking I think these recipes are become too complicated and maybe fragile.

Do you think it's better to merge all these complicated steps into a shell script? For instance, there's no need to `update-version-list` to be a target on its own. It's part of updating the charts...

Reply:
The update-version-list target must be performed before the controller-gen target, because the former updates the comments in the go code (in istio_types.go), which are then parsed by controller-gen.